mysql/resultset_helper.go

```diff
@@ -1,13 +1,54 @@
 package mysql
 
 import (
+	"bytes"
+	"encoding/binary"
 	"math"
 	"strconv"
+	"time"
 
 	"github.com/pingcap/errors"
 	"github.com/siddontang/go/hack"
 )
 
+func toBinaryDateTime(t time.Time) ([]byte, error) {
```

Comment:
There's `FormatBinaryDateTime` under `mysql/util.go`. Can you move this function adjacent to `FormatBinaryDateTime`? So we can compare their logic easier.

Also, please add a unit test for `FormatBinaryDateTime` and `ToBinaryDateTime`, but covert some value back and forth